ocs_ci/deployment/cnv.py

```diff
@@ -792,3 +797,92 @@ def uninstall_cnv(self, check_cnv_installed=True):
 
         logger.info("Removing the openshift virtualization CRDs")
         self.remove_crds()
+
+    def cnv_patch_subscription(self, patch):
+        patch_cmd = (
+            f"oc -n {self.namespace} patch sub {constants.KUBEVIRT_HYPERCONVERGED} "
+            f"-p {patch} --type merge"
+        )
+        run_cmd(patch_cmd)
+
+    def get_running_cnv_version(self):
+        """
+        Get the currently deployed cnv version
+
+        Returns:
+            string: cnv version
+
+        """
+        occmd = (
+            f"oc get sub {constants.KUBEVIRT_HYPERCONVERGED} -n openshift-cnv -o json"
+        )
+        jq_cmd = "jq -r .status.currentCSV"
+        json_out = subprocess.Popen(shlex.split(occmd), stdout=subprocess.PIPE)
+        cnv_version = subprocess.Popen(
+            shlex.split(jq_cmd), stdin=json_out.stdout, stdout=subprocess.PIPE
+        )
+        json_out.stdout.close()
+        return cnv_version.communicate()[0].decode()
+
+    def check_cnv_is_upgradable(self):
+        """
+        This method checks if the cnv operator is upgradable or not
+        """
+        logger.info("Check cnv is installed")
+        assert self.cnv_hyperconverged_installed(), "cnv operator is not installed"
+
+        logger.info("Check cnv is READY")
+        assert self.post_install_verification(
+            raise_exception=False
+        ), "CNV operator is not ready"
+
+        cmd = (
+            f"oc get hyperconverged {constants.KUBEVIRT_HYPERCONVERGED} -n openshift-cnv -o json | "
```

Comment:
Using class OCP or OCS makes the code maintainable. There are other places as well in this PR where "oc" command is executed directly.

For example,

```
kubevirt_hyperconverged = OCP(kind=constants.HYPERCONVERGED, namespace=self.namespace, resource_name=constants.KUBEVIRT_HYPERCONVERGED)
 hyperconverged_conditions = kubevirt_hyperconverged.get()["status"]["conditions"]
for condition in hyperconverged_conditions:
      if condition["type"] == "Upgradeable":
            cnv_upgradable = True if condition["status"] == "True" else False
            break
return cnv_upgradable
```

Please try to use class OCP or OCS in other places.

Reply:
Addressed.